db/seeds.rb

```diff
@@ -1,187 +1,169 @@
-providence = Location.find_or_initialize_by(
-  city_name: 'Providence'
-) do |r|
+def update_or_new(klass, attributes)
+  my_record = klass.find_by(attributes) || klass.new(attributes)
```

Comment:
nit: perhaps klass could be renamed to `record_class`? Looks good to me otherwise

Reply:
'class' is a reserved keyword for defining a class in ruby. Because it cannot be used as a variable name, the ruby community has taken to using 'klass'. Not for any special reason, but I believe it's almost a kind of naming convention at this point.

Reply:
@bbazyk @tucker-weed Ben is correct.  Because of reserved word status, this is the idiomatic way of referencing a class in code. I like your instincts that this is harder to read than using a real word and something more descriptive though.  If it wasn't fairly entrenched (you can find lots of examples if you dig around the rails codebase) it would be a solid suggestion.